src/imex.rs

```diff
@@ -512,120 +622,70 @@ async fn import_backup(context: &Context, backup_to_import: impl AsRef<Path>) ->
 /*******************************************************************************
  * Export backup
  ******************************************************************************/
-/* the FILE_PROGRESS macro calls the callback with the permille of files processed.
-The macro avoids weird values of 0% or 100% while still working. */
+#[allow(unused)]
 async fn export_backup(context: &Context, dir: impl AsRef<Path>) -> Result<()> {
     // get a fine backup file name (the name includes the date so that multiple backup instances are possible)
-    // FIXME: we should write to a temporary file first and rename it on success. this would guarantee the backup is complete.
-    // let dest_path_filename = dc_get_next_backup_file(context, dir, res);
     let now = time();
-    let dest_path_filename = dc_get_next_backup_path(dir, now).await?;
-    let dest_path_string = dest_path_filename.to_string_lossy().to_string();
+    let (temp_path, dest_path) = dc_get_next_backup_path_new(dir, now).await?;
 
+    context
+        .sql
+        .set_raw_config_int(context, "backup_time", now as i32)
+        .await?;
     sql::housekeeping(context).await;
 
     context.sql.execute("VACUUM;", paramsv![]).await.ok();
```

Comment:
Could you add a `warn` logging if vacuum fails?

Reply:
Done, but there was no warning shown when I executed this.

Reply:
I have this error: `DeltaChat: src/imex.rs:641: Vacuum failed, exporting anyway Sqlite Error: SqliteFailure(Error { code: SystemIOFailure, extended_code: 6410 }, Some("disk I/O error"))` Most likely internal sdcard is out of space.

Reply:
Strange, I freed up some space, it still fails. @Hocuri  Have you tested it on Android?

Reply:
Seems related: https://github.com/SRombauts/SQLiteCpp/issues/128

Reply:
Yes, it worked fine on my Android phone.

Reply:
But I didn't export-import before, so it might be that SQLite decided that Vacuum is not necessary.

Reply:
I still failed to reproduce this, could you check if this if fixed now? (in ce634bc9)